src/views/__init__.py

```diff
@@ -1,8 +1,16 @@
 from flask import Response, json
+from ..models.ApiKey import ApiKey
 
 def custom_response(res, status_code):
   return Response(
     mimetype="application/json",
     response=json.dumps(res),
     status=status_code
-  )
\ No newline at end of file
+  )
+
+
+def api_key_check(req):
+  if 'api-key' not in req.headers:
+    return custom_response({"error": "include an 'api-key' in request header"}, 400)
+  elif not ApiKey.get_by_key(req.headers["api-key"]):
+    return custom_response({"error": "incorrect API key"}, 400)
```

Comment:
Check if api-key was included in the request headers. Then check if there is an entry with that key. If all good, dont return anything